content/en/blog/_posts/2025-07-28-kubernetes-v1.34-sneak-peek.md

```diff
@@ -0,0 +1,97 @@
+
+To address the longstanding challenge of debugging node-level issues by correlating disconnected logs, the Kubelet OpenTelemetry Tracing enhancement ([KEP-2831](https://kep.k8s.io/2831)) provides deep, contextual insights into the `kubelet`. It instruments the `kubelet`'s critical control loops and interfaces, most notably the gRPC calls made via the Container Runtime Interface (CRI). By generating and exporting distributed trace data using the vendor-agnostic OpenTelemetry standard, it fundamentally transforms node-level diagnostics from parsing disparate log files to analyzing a single, cohesive trace. This allows operators to visualize the entire lifecycle of critical operations like Pod startup, precisely pinpointing sources of latency and errors. Its most powerful aspect is the propagation of trace context; the `kubelet` passes a trace ID with its requests to the container runtime, enabling runtimes like containerd or CRI-O to link their own spans.  
+This effort is complemented by a parallel enhancement, [KEP-647](https://kep.k8s.io/647), which brings the same tracing capabilities to the Kubernetes API server. Together, these enhancements provide a more unified, end-to-end view of events, simplifying the process of pinpointing latency and errors from the control plane down to the node. These features have matured through the official Kubernetes release process. [KEP-2831](https://kep.k8s.io/2831) was introduced as an alpha feature in v1.25, while [KEP-647](https://kep.k8s.io/647) debuted as alpha in v1.22. Both enhancements were promoted to beta together in the v1.27 release. Looking forward, Kubelet Tracing ([KEP-2831](https://kep.k8s.io/2831)) is now targeting graduation to a stable in the upcoming v1.34 release.
+
+### `PreferSameZone` and `PreferSameNode` traffic distribution for Services
+
+The `spec.trafficDistribution` field within a Kubernetes Service allows users to express preferences for how traffic should be routed to Service endpoints.  
+[KEP-3015](https://kep.k8s.io/3015) deprecates `PreferClose` and introduces two additional values: `PreferSameZone` and `PreferSameNode`. `PreferSameZone` is equivalent to the current `PreferClose`. `PreferSameNode` prioritizes sending traffic to endpoints on the same node as the client.  
+This feature was introduced in v1.33 behind the `PreferSameTrafficDistribution` feature gate. It is targeting graduation to beta in v1.34 with its feature gate enabled by default.
+
+### `kubectl` supports KYAML as an output format
+
+[KEP-5295](https://kep.k8s.io/5295) introduces KYAML, a safer and less ambiguous YAML subset and encoding, designed specifically for Kubernetes. KYAML addresses the shortcomings of both YAML and JSON. YAML's significant whitespace requires careful attention to indentation and nesting, while its optional string-quoting can lead to unexpected type coercion. Meanwhile, JSON lacks comment support and has strict requirements for trailing commas and quoted keys.  
```

Comment:
"some of the shortcomings" -- this is contentious. :)

Reply:
I see that now "some" has been removed. 

I think Tim is talking about mentioning the specific shortcomings rather than saying "some" in this context. There may be more shortcomings (again, debatable) of yaml that is not being addressed by kyaml. While we can not mention all the details in the blog, ["The Norway Bug"](https://hitchdev.com/strictyaml/why/implicit-typing-removed/) is worth mentioning. 

`
YAML's significant whitespace requires careful attention to indentation and nesting, while its optional string-quoting can lead to unexpected type coercion (for e.g.:  ["The Norway Bug"](https://hitchdev.com/strictyaml/why/implicit-typing-removed/))
`